orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_data_files.py

```diff
@@ -1,12 +1,236 @@
+    out_path = f"{out_prefix}/data-transfer-requests-deduped/*"
+
+    job_config = bigquery.job.ExtractJobConfig()
+    job_config.destination_format = bigquery.DestinationFormat.NEWLINE_DELIMITED_JSON
+    extract_job: bigquery.ExtractJob = bigquery_client.extract_table(
+        f"{scratch_dataset_name}.{file_load_table_name}",
+        destination_uris=[
+            f"{out_path}"
+        ],
+        job_config=job_config,
+        location="US",
+        project=scratch_config.scratch_bq_project
+    )
+    return extract_job.result()
+
 
-from hca_orchestration.solids.load_hca.stage_data import HcaStagingDatasetName
+@solid(
+    required_resource_keys={"data_repo_client", "scratch_config", "load_tag", "target_hca_dataset", "storage_client"},
+    input_defs=[InputDefinition("control_file_path", str)]
+)
+def run_bulk_file_ingest(context: AbstractComputeExecutionContext, control_file_path: str) -> JobId:
+    """
+    Submits the given control for ingestion to TDR
+    :param context: Dagster solid context
+    :param control_file_path: Path to the control file for ingest
+    :return: Jade Job ID
+    """
+    profile_id = context.resources.target_hca_dataset.billing_profile_id
+    dataset_id = context.resources.target_hca_dataset.dataset_id
+    scratch_bucket_name = context.resources.scratch_config.scratch_bucket_name
+
+    payload = {
+        "profileId": profile_id,
+        "loadControlFile": f"gs://{scratch_bucket_name}/{control_file_path}",
+        "loadTag": context.resources.load_tag,
+        "maxFailedFileLoads": 0
+    }
+    context.log.info(f'Bulk file ingest payload = {payload}')
+
+    data_repo_client = context.resources.data_repo_client
+    job_response: JobModel = data_repo_client.bulk_file_load(
+        dataset_id,
+        bulk_file_load=payload
+    )
+    context.log.info(f"bulk file ingest job id = {job_response.id}")
+    result: JobId = JobId(job_response.id)
+
+    return result
 
 
 @solid(
-    input_defs=[InputDefinition("staging_dataset_name", HcaStagingDatasetName)]
+    required_resource_keys={"data_repo_client"},
+    config_schema={
+        'max_wait_time_seconds': Int,
+        'poll_interval_seconds': Int,
+    }
+)
+def _base_check_bulk_file_ingest_job_result(context: AbstractComputeExecutionContext, job_id: JobId) -> Nothing:
+    # we need to poll on the endpoint as a workaround for a race condition in TDR (DR-1791)
+    job_results = api_request_with_retry(
+        context.resources.data_repo_client.retrieve_job_result,
+        context.solid_config['max_wait_time_seconds'],
+        context.solid_config['poll_interval_seconds'],
+        {500},
+        job_id
+    )
+    failed_files = job_results['failedFiles']
+    if failed_files > 0:
+        raise Failure(f"Bulk file load (job_id = {job_id} had failedFiles = {failed_files}")
+
+
+@configured(_base_check_bulk_file_ingest_job_result)
+def check_bulk_file_ingest_job_result(config: DagsterConfigDict) -> DagsterConfigDict:
```

Comment:
let's generally avoid directly calling `@configured` in favor of using `preconfigure_for_mode` for passing in all default configs like this

Reply:
@ThePletch how would that work for a solid in this context? I'm unsure how I'd pull in the mode name

Reply:
hm, good point. it seems like the correct approach here is to use a preset, but we'd need to have a think about best practices for presets if we wanted to do that, as well as consider whether there'd be other downsides.